airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3StorageOperations.java

```diff
@@ -130,7 +132,13 @@ public String uploadRecordsToBucket(final SerializableBuffer recordsData,
         exceptionsThrown.add(e);
       }
     }
-    throw new RuntimeException(String.format("Exceptions thrown while uploading records into storage: %s", Strings.join(exceptionsThrown, "\n")));
+    final boolean areAllExceptionsAuthExceptions = exceptionsThrown.stream().filter(e -> e instanceof AmazonS3Exception)
```

Comment:
if I'm reading this correctly, this currently checks that all `AmazonS3Exception` are auth errors, but ignores any non-amazon exceptions. E.g. if there's a random `IllegalStateException` in this list, it would be treated as though it's an auth error

I think it might be better to bundle the filter + map together?
```java
.map(e -> {
  if (e instanceof AmazonS3Exception s3e) {
    // ...
  } else {
    // assume non-amazon exceptions are unrelated to authentication
    return false;
  }
})
```

Reply:
It took me awhile but I think I see what you and @grishick are saying - updating this to ensure that really all of the errors are auth errors